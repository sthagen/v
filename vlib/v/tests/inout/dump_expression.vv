fn dump_of_int() {
	x := dump(1) + 1
	assert x == 2
}

fn dump_of_string() {
	x := dump('a') + 'b'
	assert x == 'ab'
}

struct Point {
mut:
	x int
	y int
	z int
}

fn dump_of_struct() {
	point := Point{1, 2, 3}
	mut x := dump(point)
	x.x += 100
	assert x == Point{101, 2, 3}
}

fn main() {
	dump_of_int()
	dump_of_string()
	dump_of_struct()
}
