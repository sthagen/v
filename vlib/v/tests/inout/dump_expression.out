[vlib/v/tests/inout/dump_expression.vv:2] 1: 1
[vlib/v/tests/inout/dump_expression.vv:7] 'a': a
[vlib/v/tests/inout/dump_expression.vv:20] point: Point{
    x: 1
    y: 2
    z: 3
}
